opslevel/datasource_opslevel_service_dependencies.go

```diff
@@ -0,0 +1,168 @@
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+// Ensure ServiceDataSource implements DataSourceWithConfigure interface
+var _ datasource.DataSourceWithConfigure = &ServiceDataSource{}
+
+func NewServiceDependenciesDataSource() datasource.DataSource {
+	return &ServiceDependenciesDataSource{}
+}
+
+// ServiceDataSource manages a Service data source.
+type ServiceDependenciesDataSource struct {
+	CommonDataSourceClient
+}
+
+// ServiceDependenciesModel describes the data source data model.
+type ServiceDependenciesModel struct {
+	Dependents   []dependentsModel   `tfsdk:"dependents"`
+	Dependencies []dependenciesModel `tfsdk:"dependencies"`
+	Service      types.String        `tfsdk:"service"`
+}
+
+type dependentsModel struct {
+	Id     types.String `tfsdk:"id"`
+	Locked types.Bool   `tfsdk:"locked"`
+	Notes  types.String `tfsdk:"notes"`
+}
+
+type dependenciesModel struct {
+	Id     types.String `tfsdk:"id"`
+	Locked types.Bool   `tfsdk:"locked"`
+	Notes  types.String `tfsdk:"notes"`
+}
+
+func NewServiceDependenciesModel(serviceIdentifier string, dependents []dependentsModel, dependencies []dependenciesModel) ServiceDependenciesModel {
+	return ServiceDependenciesModel{
+		Dependents:   dependents,
+		Dependencies: dependencies,
+		Service:      types.StringValue(serviceIdentifier),
+	}
+}
+
+func (d *ServiceDependenciesDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_dependencies"
+}
+
+var depsAttrs = map[string]schema.Attribute{
+	"id": schema.StringAttribute{
+		Description: "The ID of the serviceDependency.",
+		Computed:    true,
+	},
+	"locked": schema.BoolAttribute{
+		Description: "Is the dependency locked by a service config?",
+		Computed:    true,
+	},
+	"notes": schema.StringAttribute{
+		Description: "Notes for service dependency.",
+		Optional:    true,
+	},
+}
+
+func (d *ServiceDependenciesDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Service Dependencies data source",
+
+		Attributes: map[string]schema.Attribute{
+			"dependents": schema.ListNestedAttribute{
+				Description: "List of Service Dependents of a service",
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: depsAttrs,
+				},
+				Computed: true,
+			},
+			"dependencies": schema.ListNestedAttribute{
+				Description: "List of Service Dependencies of a service",
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: depsAttrs,
+				},
+				Computed: true,
+			},
+			"service": schema.StringAttribute{
+				Description: "The ID or alias of the service with the dependency.",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func (d *ServiceDependenciesDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var (
```

Comment:
Recommend we don't follow the pattern of declaring `var` at the top of the function